include/nvs/nvs.h

```diff
@@ -0,0 +1,224 @@
+/*  NVS: non volatile storage in flash
+ *
+ * Copyright (c) 2018 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __NVS_H_
+#define __NVS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <sys/types.h>
+
+#define NVS_MOVE_BLOCK_SIZE 8
+
+/*
+ * Error codes.
+ */
+#define NVS_OK		 0
```

Comment:
Remove those and use our standard error codes

Reply:
Hi Carles, I guess you mean using errno.h, this does not contain the necessary error codes for the following defines:

``#define NVS_ERR_ARGS	-1
#define NVS_ERR_FLASH	-2
#define NVS_ERR_NOVAR   -3
#define NVS_ERR_NOSPACE	-4
#define NVS_ERR_MAGIC   -7
#define NVS_ERR_CRC     -8
#define NVS_ERR_CFG     -9
#define NVS_ERR_LEN     -10``

Also fcb and nffs are using their own error codes, so why should NVS be the exception?

Reply:
@carlescufi Should I use standard error codes or not ?

Reply:
Apologies for not replying earlier. I would prefer that you use standard error codes even if they are not as descriptive. FCB and NFFS use their own error codes because they come from external projects, but, at least in the FCB case, I do want to get rid of those as well.

Reply:
@carlescufi I think reusing standard error codes being less descriptive seems like a bad idea. However I do think there is something wrong with the error codes as they are defined in nvs, fcb, nffs, ...: they are redefintions of existing error codes, this is confusing. Would it not be better to allow module specific (and descriptive) error codes but not to allow them to "overwrite" existing error codes. Suppose we would allow module specific error codes only starting with 0x7F: so module specific error codes are 0x7F00 - 0x7FFF (256 possible error codes). This allows easy distinguishing between module specific and generic error codes. What's your opinion ?

Reply:
I don't dislike the option at all, but this must be discussed in a bit of a wider scope. We have 2 libc implementations in Zephyr: minimal and newlib, so this needs to work with both. I noticed that newlib has a `ELASTERROR` defined to add user values after it, but minimal does not:

https://github.com/zephyrproject-rtos/zephyr/blob/master/lib/libc/minimal/include/errno.h

https://sourceware.org/git/gitweb.cgi?p=newlib-cygwin.git;a=blob_plain;f=newlib/libc/include/sys/errno.h;hb=HEAD

What I suggest is that you send an email to the devel mailing list asking about the canonical way to extend errno.h in Zephyr, what do you think?

Reply:
@carlescufi There seems to be little interest from the developers in the method of extending errno.h (except from really extending errno.h which would halt development if extension is needed). I have been going through other modules and maybe there is already a solution in use (e.g. bluetooth mesh): Use errno.h as much as possible with the definition of negative return values, use specific positive return values (not error but status codes) if your module needs it. Don't define a OK value, return 0 if OK. 
I think I can reduce the amount of specific error codes in NVS to about 3 or 4 status codes.
What do you think about this approach ?